src/transaction/transaction.c

```diff
@@ -169,19 +165,19 @@ bool checkSwapConditionsAndSign(parseResult_t *transaction) {
     return true;
 }
 
-void reviewTransaction(parseResult_t *transaction, action_t onApprove, action_t onReject) {
-    approvalAction = onApprove;
-    rejectionAction = onReject;
+void review_transaction(parseResult_t *transaction, action_t on_approve, action_t on_reject) {
+    approval_action = on_approve;
+    rejection_action = on_reject;
 
     if (called_from_swap) {
-        if (checkSwapConditionsAndSign(transaction)) {
-            approvalAction();
+        if (check_swap_conditions_and_sign(transaction)) {
+            approval_action();
```

Comment:
I haven't tried the swap functionality yet but since swap doesn't require for user approving a transaction could you in-short explain how swap logical flow goes?

Reply:
First, read this: https://blog.ledger.com/secure-swap/
Basically, after a user approval, the Exchange app jumps into the coin app with preapproved address and amounts, then if a transaction streamed to the coin app matches these parameters, it gets signed without user consent.

Reply:
Thank you for the explanation. I've missed the exchange app in the process.
Btw is the protocol for communication between the apps on ledger device new?

Reply:
There's no "protocol", it's a syscall to jump from an app to a library (another app authorized to be jumped in). In this case, both the XRP and Exchange app are libraries.
The first libraries in our ecosystem have been BTC/ETH, to save space for all their clones, something like 2 years ago when firmware 1.4.2 came out on Nano S.

Reply:
Ok understand. I didn't know you could syscall into another app. 
Thx for all the information!